build/targets/standard-lifecycle.targets

```diff
@@ -1,57 +1,153 @@
+
+  <Target Name="CleanArtifacts">
+    <!-- intentionally BuildDir, not ArtifactDir -->
     <RemoveDir Directories="$(BuildDir)" />
+  </Target>
+
+<!--
+###################################################################
+Targets: {Target}Solutions
+
+Items: Solutions
+
+Executes /t:{Target} on all solutions
+###################################################################
+-->
+  <Target Name="CleanSolutions">
     <MSBuild Targets="Clean"
-              Projects="@(Solutions)"
-              Properties="Configuration=$(Configuration)" />
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
   </Target>
 
-  <Target Name="Restore">
+  <Target Name="RestoreSolutions">
     <MSBuild Targets="Restore"
-             Projects="@(Solutions)"
-             Properties="Configuration=$(Configuration)" />
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
   </Target>
 
-  <Target Name="Compile">
+  <Target Name="BuildSolutions">
     <MSBuild Targets="Build"
-             Projects="@(Solutions)"
-             Properties="Configuration=$(Configuration)" />
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
   </Target>
 
-  <Target Name="Package">
-    <!-- TODO consider using GeneratePackageOnBuild or calling Pack on the solution. Requires setting IsPackable=false on test and samples -->
+  <Target Name="RebuildSolutions">
+    <MSBuild Targets="Rebuild"
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
+  </Target>
+
+<!--
+###################################################################
+Target: PackageProjects
+
+Executes /t:Pack on all projects matching src/*/*.csproj.
+###################################################################
+-->
+  <Target Name="PackageProjects">
+    <!-- TODO consider using GeneratePackageOnBuild or calling Pack on the solution.
+        This would require setting IsPackable=false and samples and possible on tests (see https://github.com/NuGet/NuGet.Client/pull/1170) -->
     <ItemGroup>
       <ProjectsToPack Include="$(RepositoryRoot)src\*\*.csproj" />
     </ItemGroup>
+
     <MSBuild Targets="Pack"
-             Projects="@(ProjectsToPack)"
-             Properties="Configuration=$(Configuration);PackageOutputPath=$(BuildDir)" />
+      Projects="@(ProjectsToPack)"
+      Properties="Configuration=$(Configuration);PackageOutputPath=$(BuildDir)" />
   </Target>
 
-  <!-- TODO -->
-  <Target Name="Test" />
+<!--
+###################################################################
+Target: PackSharedSources
 
-  <Target Name="Verify" DependsOnTargets="VerifyPackages" />
+Creates a content files package for all each directory in
+that matches "$(RepositoryRoot)/shared/*.Sources".
+###################################################################
+-->
+  <Target Name="PackSharedSources">
+    <!-- TODO consider using GeneratePackageOnBuild or calling Pack on the solution.
```

Comment:
This wouldt't apply for shared source projects, would it? Unless we intend on adding a project per shared src project.